interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentExtractor.scala

```diff
@@ -80,7 +80,7 @@ case class ComponentExtractor(classLoader: ClassLoader, nussknackerVersion: Nuss
   private def extractOneProviderConfig(config: ComponentProviderConfig, provider: ComponentProvider, processObjectDependencies: ProcessObjectDependencies): List[(String, WithCategories[Component])] = {
     provider.create(config.config, processObjectDependencies).map { cd =>
       val finalName = config.componentPrefix.map(_ + cd.name).getOrElse(cd.name)
-      finalName -> WithCategories(cd.component, config.categories, SingleComponentConfig.zero.copy(docsUrl = cd.docsUrl, icon = cd.icon))
+      finalName -> WithCategories(cd.component, config.categories, SingleComponentConfig.zero.copy(docsUrl = cd.docsUrl, icon = cd.icon, componentId = Some(ComponentId.create(finalName))))
```

Comment:
Ahh I've take a look into how those components are configured and it is not that way as I was thinking. Take a look at `dev-application.conf` - they are defined for each `scenarioType`. I'm thinking about passing there `None` and let's user define componentId in `componentsUiConfig`. But I see now one problem with `componentsUiConfig`. It is map `label -> config` and it should be rather `(label, componentType) -> config` to make sure that `kafka-avro` source and sink will work together. But maybe we can left it how it is and think about it when this problem appear?